rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/services/ServiceConfig.java

```diff
@@ -51,19 +60,50 @@ RetrySupport retrySupport() {
   @Bean
   ClouddriverService clouddriverService(
       Ok3Client ok3Client, RequestInterceptor spinnakerRequestInterceptor) {
-    ObjectMapper objectMapper =
-        new ObjectMapper()
-            .enable(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_AS_NULL)
-            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
 
     return new RestAdapter.Builder()
         .setEndpoint(clouddriverBaseUrl)
         .setRequestInterceptor(spinnakerRequestInterceptor)
         .setClient(ok3Client)
-        .setConverter(new JacksonConverter(objectMapper))
+        .setConverter(new JacksonConverter(getObjectMapper()))
         .setLogLevel(RestAdapter.LogLevel.valueOf(retrofitLogLevel))
         .setErrorHandler(SpinnakerRetrofitErrorHandler.getInstance())
         .build()
         .create(ClouddriverService.class);
   }
+
+  private ObjectMapper getObjectMapper() {
+    return new ObjectMapper()
+        .enable(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_AS_NULL)
+        .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+  }
+
+  /*As part of retrofit2 changes, creating clouddriverservice with retrofit2 API changes */
+  @Bean
+  ClouddriverRetrofit2Service clouddriverRetrofit2Service(
+      OkHttp3ClientConfiguration okHttpClientConfig,
+      SpinnakerRequestHeaderInterceptor spinnakerRequestHeaderInterceptor) {
+    /*.
+     * As part this OkHttpClient builder  OkHttp3MetricsInterceptor added as first interceptor, but this interceptor needs spinnaker
```

Comment:
```suggestion
     * As part of okHttpClientBuilder, OkHttp3MetricsInterceptor is the first interceptor, but this interceptor needs spinnaker
```

Reply:
I still think some tweaks here would help.

Reply:
As I think about it more though, we're going to need code like this in every service that uses retrofit2, right?  What do think about adding a method like `createForRetrofit2` (or maybe there's a better name) to [OkHttp3ClientConfiguration](https://github.com/spinnaker/kork/blob/b6fa1103b7b3b12734af39cb6f6ca2b04cdd59d6/kork-web/src/main/groovy/com/netflix/spinnaker/config/OkHttp3ClientConfiguration.groovy)?

I am curious how this works with retrofit1.  Like, OkHttp3MetricsInterceptor still needs headers to do it's work (e.g. [here](https://github.com/spinnaker/kork/blob/b6fa1103b7b3b12734af39cb6f6ca2b04cdd59d6/kork-web/src/main/java/com/netflix/spinnaker/okhttp/MetricsInterceptor.java#L120])..but how does that happen?   Like, how can we get away with the existing [create](https://github.com/spinnaker/kork/blob/b6fa1103b7b3b12734af39cb6f6ca2b04cdd59d6/kork-web/src/main/groovy/com/netflix/spinnaker/config/OkHttp3ClientConfiguration.groovy#L63) method not adding [SpinnakerRequestInterceptor](https://github.com/spinnaker/kork/blob/b6fa1103b7b3b12734af39cb6f6ca2b04cdd59d6/kork-web/src/main/groovy/com/netflix/spinnaker/okhttp/SpinnakerRequestInterceptor.groovy) ?

Reply:
In retrofit1 SpinnakerRequestInterceptor was executed initially later the okhttp3 interceptors are executed, so headers are already present for interceptors...

Reply:
ya, it is better to centralize, so we can use the same in other services too...

Reply:
According to the documentation of RequestInterceptor, it will execute before every request is made.
https://www.javadoc.io/doc/com.squareup.retrofit/retrofit/1.9.0/retrofit/RequestInterceptor.html